pkg/trace/api/profiles.go

```diff
@@ -68,7 +74,20 @@ func (r *HTTPReceiver) profileProxyHandler() http.Handler {
 	if err != nil {
 		return errorHandler(err)
 	}
-	tags := fmt.Sprintf("host:%s,default_env:%s", r.conf.Hostname, r.conf.DefaultEnv)
+	ver, err := version.Agent()
+	verStr := "unknown"
+	if err == nil {
+		verStr = ver.GetNumberAndPre()
+	}
+
+	tags := fmt.Sprintf("host:%s,default_env:%s,agent_version:%s", r.conf.Hostname, r.conf.DefaultEnv, verStr)
+
+	// Note: check for Fargate explicitly, need identifying tags for host tracking
+	if fargate.IsFargateInstance(context.Background()) {
+		orchestratorTag := fmt.Sprintf("orchestrator:fargate_%s",
+			strings.ToLower(string(fargate.GetOrchestrator(context.Background()))))
```

Comment:
You should use a request with an exaggerated timeout of 2 seconds, just to be on the safe side. Something like this (with a different error message): https://github.com/DataDog/datadog-agent/blob/f9f4d62abf41ae753a5477b498479d1bca0ef7a2/pkg/trace/config/config.go#L143-L148

Reply:
If i use `r.conf.IsFargate` as suggested above, I don't need this at all

Reply:
I was referring to the `fargate.GetOrchestrator(context.Background())` part.